pkg/tsdb/cloudwatch/cloudwatch.go

```diff
@@ -324,18 +344,19 @@ func (e *cloudWatchExecutor) QueryData(ctx context.Context, req *backend.QueryDa
 		frontend, but because alerts are executed on the backend the logic needs to be reimplemented here.
 	*/
 	q := req.Queries[0]
-	model, err := simplejson.NewJson(q.JSON)
+	var model DataQueryJson
```

Comment:
Seems like `DataQuery` is only used in annotation queries. I think it would be good to highlight the difference between the three different query types in a similar way to how it's done in the [frontend](https://github.com/grafana/grafana/blob/cloudwatch-remove-simplejson/public/app/plugins/datasource/cloudwatch/types.ts#L104). In the `executeAnnotationQuery` you can unmarshal json into an `AnnotationDataQuery` struct. In `executeLogActions`, you can unmarshal into a LogsDataQuery and in `executeTimeSeriesQuery` you can unmarshal into a `MetricsDataQuery`.

Reply:
Oh and `executeLogAlertQuery` is somehow in `cloudwatch.go` :man_shrugging: Which seems odd as well...